ClientApp/viking-vault/src/display_user_list/UserData.tsx

```diff
@@ -24,59 +34,96 @@ const defaultUser = {
     email: "no-data",
     pictureLink: "",
     cardNumber: "no-data",
-    expirationDate: "no-data"
+    expirationDate: "no-data",
+    openDeleteUserModal: false
 };
 
-class UserData extends React.Component<IUserData, IUserData>{
+class UserData extends React.Component<IUserDataProp, IPageState>{
     
-    constructor(props: IUserData)
+    constructor(props: IUserDataProp)
     {
         super(props);
 
         if(props !== null)
         {
             this.state = {
-                id: props.id,
-                firstName: props.firstName,
-                lastName: props.lastName,
-                address: props.address,
-                email: props.email,
-                pictureLink: props.pictureLink,
-                cardNumber: props.cardNumber,
-                expirationDate: props.expirationDate
+                user : {
+                    id: props.user.id,
+                    firstName: props.user.firstName,
+                    lastName: props.user.lastName,
+                    address: props.user.address,
+                    email: props.user.email,
+                    pictureLink: props.user.pictureLink,
+                    cardNumber: props.user.cardNumber,
+                    expirationDate: props.user.expirationDate
+                },
+                openDeleteUserModal: false
             };
         }
         else
         {
-            this.state = {...defaultUser};
+            this.state = {
+                user : {...defaultUser},
+                openDeleteUserModal : false
+            };
         }        
     }
 
     renderUserCard()
     {
-        if(this.state.cardNumber !== "")
+        if(this.state.user.cardNumber !== "")
             return <div className = "card-data-style">
-                        <p className = "name-on-card">{this.state.firstName} {this.state.lastName}</p>
-                        <p className = "expiration-date-on-card">{this.state.expirationDate}</p>
-                        <p className = "card-number-on-card">{this.state.cardNumber}</p>
+                        <p className = "name-on-card">{this.state.user.firstName} {this.state.user.lastName}</p>
+                        <p className = "expiration-date-on-card">{this.state.user.expirationDate}</p>
+                        <p className = "card-number-on-card">{this.state.user.cardNumber}</p>
                         <img src = {CardImg} className = "card-img"></img>
                    </div>        
 
         return null; 
     }
 
+    private handleDeleteUser = () =>{
+        this.setState((oldstate : any)=>({
```

Comment:
why do you use any for oldState? (notice the naming)

Reply:
That was the only way I managed to make this work, by specifying any for the type of oldState. Anyhow, this was needed in the early version of this feature, but I removed it since we no longer need to toggle the state.